spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -155,6 +158,20 @@ public long count(Class<?> entityClass) {
 		return findById(id, entityClass) != null;
 	}
 
+	@Override
+	public <A> A performTransaction(Function<DatastoreOperations, A> operations) {
+		if (!(this.datastore instanceof Datastore)) {
+			throw new DatastoreDataException(
+					"This DatastoreReadWriter cannot be used to run transactions. A full Datastore service"
+							+ " object is required to run functions as transactions.");
+		}
+		return ((Datastore) this.datastore).runInTransaction(
+				readerWriter -> operations.apply(new DatastoreTemplate(readerWriter,
```

Comment:
please add type specifier for readerWriter

Reply:
Could you clarify what you mean? The anonymous function here is an implementation for the one-function interface TransactionCallable. The type of `readWriter` is determined by that `TransactionCallable` interface.

Reply:
I meant `(DatastoreReaderWriter readerWriter) -> ...`

Reply:
Change made, PTAL